tests/src/python/test_provider_ogr_general.py

```diff
@@ -0,0 +1,980 @@
+import sys
+import tempfile
+
+from qgis.core import Qgis, QgsVectorLayer, QgsVectorDataProvider, QgsWkbTypes, QgsFeatureRequest, QgsGeometry
+from qgis.PyQt.QtCore import QDate
+from qgis.testing import (
+    start_app,
+    unittest
+)
+from utilities import unitTestDataPath
+from osgeo import gdal, osr, ogr
+from inspect import getmembers
+from pprint import pprint
+
+start_app()
+TEST_DATA_DIR = unitTestDataPath()
+
+
+def GDAL_COMPUTE_VERSION(maj, min, rev):
+    return ((maj) * 1000000 + (min) * 10000 + (rev) * 100)
+
+# Note - doesn't implement ProviderTestCase as most OGR provider is tested by the shapefile provider test
+
+
+def count_opened_filedescriptors(filename_to_test):
+    count = -1
+    if sys.platform.startswith('linux'):
+        count = 0
+        open_files_dirname = '/proc/%d/fd' % os.getpid()
+        filenames = os.listdir(open_files_dirname)
+        for filename in filenames:
+            full_filename = open_files_dirname + '/' + filename
+            if os.path.exists(full_filename):
+                link = os.readlink(full_filename)
+                if os.path.basename(link) == os.path.basename(filename_to_test):
+                    count += 1
+    return count
+
+
+class TestPyQgsOGRProviderGeneral(unittest.TestCase):
+
+    ogr_spatialview_insert=0
+    ogr_spatialview_update=0
+    ogr_spatialview_delete=0
+    ogr_spatialview_selectatid=0
+    ogr_spatialview_changegeonetries=0
+    gdal_version_num=0
+    gdal_build_num=0
+    gdal_build_version=""
+    ogr_runtime_supported=1
+
+    def setUp(self):
+        """Run before each test."""
+        self.gdal_version_num = int(gdal.VersionInfo('VERSION_NUM'))
+        try:
+            Qgis.GDAL_BUILD_VERSION
+        except AttributeError:
+            self.gdal_build_num = self.gdal_version_num
+            self.gdal_build_version = gdal.VersionInfo('VERSION_NUM')
+        else:
+            self.ogr_runtime_supported = Qgis.GDAL_OGR_RUNTIME_SUPPORTED
+            self.gdal_build_version = Qgis.GDAL_BUILD_VERSION
+            self.gdal_build_num = (Qgis.GDAL_BUILD_VERSION_MAJOR*1000000)+(Qgis.GDAL_BUILD_VERSION_MINOR*10000)+(Qgis.GDAL_BUILD_VERSION_REV*100)
+
+
+    @classmethod
+    def setUpClass(cls):
+        """Run before all tests"""
+
+
+        # Create test layer
+        cls.basetestpath = tempfile.mkdtemp()
+        cls.dirs_to_cleanup = [cls.basetestpath]
+
+    @classmethod
+    def tearDownClass(cls):
+        """Run after all tests"""
+        for dirname in cls.dirs_to_cleanup:
+            shutil.rmtree(dirname, True)
+
+###############################################################################
+# Test spatialite spatial views - writable -  #5582: SpatialView - writable
+# to (re)create (a possibily damage) database with:
+# spatialite gdal_220.autotest.ogr_spatialite_views_writable.sqlite < gdal_220.autotest.ogr_spatialite_views_writable.sql
+# Note: this function should run first
+#
+    def test_00_OgrSpatialiteViewsWritable(self):
```

Comment:
This is a test to check if the used gdal version supports writable Spatialite Views, based on this:
https://trac.osgeo.org/gdal/ticket/5582
Since my main work with gdal/QGIS uses a gdal version that supports this and needs to be regularly updated due to structural changes in gdal, a test is useful.
When this spatialite functionality (since version 2.4) is one day realized, a useful test is in place.